From f67a732465c863d7767cb9bfd8d71403806a14c8 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Thu, 17 Jun 2021 15:48:53 +0100
Subject: [PATCH 3/8] Add option to change eSCO Transport Unit Size

Fixes Bluetooth calls on some Samsung devices if set to 16.

It's still unknown if other Treble devices need other values so it's preferred to leave this able to be configured with another integer.

This applies to mSBC T2, T1 and CVSD codecs

Change-Id: I3e5897c2ce983042b9a4bef9def6744ba4253bcb
---
 system/device/src/esco_parameters.cc | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/system/device/src/esco_parameters.cc b/system/device/src/esco_parameters.cc
index 15f1397087..21590782a6 100644
--- a/system/device/src/esco_parameters.cc
+++ b/system/device/src/esco_parameters.cc
@@ -25,7 +25,8 @@
 #include "hci/controller_interface.h"
 #include "main/shim/entry.h"
 
-static const enh_esco_params_t default_esco_parameters[ESCO_NUM_CODECS] = {
+#include <cutils/properties.h>
+static enh_esco_params_t default_esco_parameters[ESCO_NUM_CODECS] = {
     // CVSD D1
     {
         .transmit_bandwidth = TXRX_64KBITS_RATE,
@@ -352,6 +353,11 @@ enh_esco_params_t esco_parameters_for_codec(esco_codec_t codec, bool offload) {
     }
   }
 
+  int escoTransportUnitSize = property_get_int32("persist.sys.bt.esco_transport_unit_size", 0);
+  if(escoTransportUnitSize) {
+    default_esco_parameters[codec].input_transport_unit_size = escoTransportUnitSize;
+    default_esco_parameters[codec].output_transport_unit_size = escoTransportUnitSize;
+  }
   if (offload) {
     if (codec == ESCO_CODEC_SWB_Q0 || codec == ESCO_CODEC_SWB_Q1 ||
         codec == ESCO_CODEC_SWB_Q2 || codec == ESCO_CODEC_SWB_Q3) {
-- 
2.34.1

